ada_2_fhir-r4/zibs2020/payload/0.0/nl-core-AlcoholUse.xsl

```diff
@@ -0,0 +1,150 @@
+                        <xsl:call-template name="code-to-CodeableConcept"/>
+                    </valueCodeableConcept>
+                </xsl:for-each>
+                             
+               <xsl:for-each select="toelichting">
+                    <note>
+                        <text>
+                            <xsl:attribute name="value" select="./@value"/>
+                        </text>
+                    </note>
+                </xsl:for-each>
+                
+                <xsl:for-each select="waarneming_gebruik/hoeveelheid">
+                    <component>
+                        <code>
+                            <coding>
+                                <system value="http://snomed.info/sct"/>
+                                <!-- DEPRECATED CODE in SNOMED -->
+                                <code value="160573003"/>
+                                <display value="inname van alcohol"/>
+                            </coding>
+                        </code>
+                        <valueQuantity>
+                            <xsl:call-template name="hoeveelheid-to-Quantity">
+                                <xsl:with-param name="in" select="."/>
+                            </xsl:call-template>
+                        </valueQuantity	>
+                    </component>
+                </xsl:for-each>
+            </Observation>
+        </xsl:for-each>
+    </xsl:template>
+    
+    <xd:doc>
+        <xd:desc>Template to generate a unique id to identify this instance.</xd:desc>
+    </xd:doc>
+    <xsl:template match="alcohol_gebruik" mode="_generateId">
+        <xsl:variable name="parts">
+            <xsl:text>AlcoholUse</xsl:text>
+            <xsl:value-of select="alcohol_gebruik_status/@displayName"/>
+            <xsl:value-of select="waarneming_gebruik/start_datum/@value"/>
+            <xsl:value-of select="concat(waarneming_gebruik/hoeveelheid/@value, waarneming_gebruik/hoeveelheid/@unit)"/>
+            <xsl:value-of select="toelichting/@value"/>
+        </xsl:variable>
+        <xsl:value-of select="substring(replace(string-join($parts, '-'), '[^A-Za-z0-9-.]', ''), 1, 64)"/>
+    </xsl:template>
+    
+    <xd:doc>
+        <xd:desc>Template to generate a display that can be shown when referencing this instance.</xd:desc>
+    </xd:doc>
+    <xsl:template match="alcohol_gebruik" mode="_generateDisplay">
+        <xsl:variable name="parts">
+            <xsl:text>AlcoholUse</xsl:text>
+            <xsl:value-of select="alcohol_gebruik_status/@displayName"/>
+            <xsl:value-of select="concat(waarneming_gebruik/hoeveelheid/@value, waarneming_gebruik/hoeveelheid/@unit)"/>
+            <xsl:if test="waarneming_gebruik/start_datum/@value">
+                <xsl:value-of select="concat('start datum', waarneming_gebruik/start_datum/@value)"/>
+            </xsl:if>
+        </xsl:variable>
+        <xsl:value-of select="string-join($parts, ', ')"/>     
+    </xsl:template>
```

Comment:
Zie opm Tobacco

Reply:
Ik denk dat de _generateDisplay de enige aanpassing is die ik nog wel echt even zou willen zien.

Reply:
Aangepast!